src/lcm/argmax.py

```diff
@@ -0,0 +1,149 @@
+import jax
+import jax.numpy as jnp
+
+# ======================================================================================
+# argmax
+# ======================================================================================
+
+
+def argmax(a, axis=None, initial=None, where=None):
+    """Compute the argmax of a n-dim array along axis.
+
+    Args:
+        a (jax.numpy.ndarray): Multidimensional jax array.
+        axis (int | tuple | None): Axis along which to compute the argmax. If None, the
+            argmax is computed over all axes.
+        initial (scalar): The minimum value of an output element. Must be present to
+            allow computation on empty slice. See ~numpy.ufunc.reduce for details.
+        where (jax.numpy.ndarray): Boolean array of the same shape as a. Only the values
+            for which where is True are considered for the argmax.
+
+    Returns:
+        - jax.numpy.ndarray: Array with the same shape as a, except for the dimensions
+          specified in axis, which are dropped. The value corresponds to an index
+          that can be translated into a tuple of indices using jnp.unravel_index.
+
+    """
+    # Preparation
+    # ==================================================================================
+    if axis is None:
+        axis = tuple(range(a.ndim))
+    elif isinstance(axis, int):
+        axis = (axis,)
+
+    # Move axis over which to compute the argmax to the back and flatten last dims
+    # ==================================================================================
+    a = _move_axes_to_back(a, axes=axis)
+    a = _flatten_last_n_axes(a, n=len(axis))
+
+    # Do same transformation for where
+    # ==================================================================================
+    if where is not None:
+        where = _move_axes_to_back(where, axes=axis)
+        where = _flatten_last_n_axes(where, n=len(axis))
+
+    # Compute argmax over last dimension
+    # ==================================================================================
+    _max = jnp.max(a, axis=-1, keepdims=True, initial=initial, where=where)
+    max_value_mask = a == _max
+    if where is not None:
+        max_value_mask = max_value_mask & where
+    argmax = jnp.argmax(max_value_mask, axis=-1)
+
+    return argmax, _max.reshape(argmax.shape)
+
+
+def _move_axes_to_back(a, axes):
+    front_axes = tuple(set(range(a.ndim)) - set(axes))
```

Comment:
Doesn't the set mess up the order (potentially). If so, the output array could end up transposed and the translation from generalized index to normal indices would get silently produce wrong results. 

It did not happen in a simple test but is definitely dangerous and set orders are not guaranteed. 

We should also add some tests for `_move_axes_to_back` and `_flatten_last_n_axes`